packages/cli/src/util/PackageManager.js

```diff
@@ -1,77 +1,56 @@
- * LICENSE file in the root directory of this source tree.
- *
- * @format
- */
-
-import { spawnSync } from 'child_process';
+// @flow
+import { execSync } from 'child_process';
 import yarn from './yarn';
 
-const spawnOpts = {
-  stdio: 'inherit',
-  stdin: 'inherit',
+type PackageManagerOptions = {
+  forceNpm?: boolean,
+  projectDir: string,
 };
 
-/**
- * Execute npm or yarn command
- *
- * @param  {String} yarnCommand Yarn command to be executed eg. yarn add package
- * @param  {String} npmCommand  Npm command to be executed eg. npm install package
- * @param  {string} projectDir  Directory to run the command in
- * @return {object}             spawnSync's result object
- */
-function callYarnOrNpm(yarnCommand, npmCommand, projectDir) {
-  let command;
-
-  const isYarnAvailable =
-    yarn.getYarnVersionIfAvailable() && yarn.isGlobalCliUsingYarn(projectDir);
+export default class PackageManager {
+  options: PackageManagerOptions;
 
-  if (isYarnAvailable) {
-    command = yarnCommand;
-  } else {
-    command = npmCommand;
+  constructor(options: PackageManagerOptions) {
+    this.options = options;
   }
 
-  const args = command.split(' ');
-  const cmd = args.shift();
+  executeCommand(command: string) {
+    return execSync(command, { stdio: 'inherit' });
+  }
 
-  const res = spawnSync(cmd, args, spawnOpts);
+  shouldCallYarn() {
+    return (
+      !this.options.forceNpm &&
+      yarn.getYarnVersionIfAvailable() &&
+      yarn.isGlobalCliUsingYarn(this.options.projectDir)
+    );
+  }
 
-  return res;
-}
+  install(packageNames: Array<string>) {
+    if (this.shouldCallYarn()) {
+      this.executeCommand(`yarn add ${packageNames.join(' ')}`);
+    } else {
+      this.executeCommand(
+        `npm install ${packageNames.join(' ')} --save --save-exact`
+      );
+    }
+  }
 
-/**
- * Install package into project using npm or yarn if available
- * @param  {[type]} packageName Package to be installed
- * @param  {string} projectDir  Root directory of the project
- * @return {[type]}             spawnSync's result object
- */
-function add(packageName, projectDir) {
-  return callYarnOrNpm(
-    `yarn add ${packageName}`,
-    `npm install ${packageName} --save`,
-    projectDir
-  );
-}
+  installDev(packageNames: Array<string>) {
+    if (this.shouldCallYarn()) {
+      this.executeCommand(`yarn add -D ${packageNames.join(' ')}`);
+    } else {
+      this.executeCommand(
+        `npm install ${packageNames.join(' ')} --save-dev --save-exact`
+      );
+    }
+  }
 
-/**
- * Uninstall package from project using npm or yarn if available
- * @param  {[type]} packageName Package to be uninstalled
- * @param  {string} projectDir  Root directory of the project
- * @return {Object}             spawnSync's result object
- */
-function remove(packageName, projectDir) {
-  return callYarnOrNpm(
-    `yarn remove ${packageName}`,
-    `npm uninstall --save ${packageName}`,
-    projectDir
-  );
+  uninstall(packageName: string) {
```

Comment:
this should accept an array for symmetry with `install`

Reply:
True, I'll change it.